### PR TITLE
fix: [PIE-2402]: skipped stage unselection

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
##### Summary:

We should not auto select a stage that is skipped

##### Jira Links:

https://harness.atlassian.net/browse/PIE-2402

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by vkbansal — APPROVED** [review]

**Review by kapilkumawat86 — APPROVED** [review]

File: src/modules/70-pipeline/utils/executionUtils.ts [hunk 1]
```diff
@@ -216,6 +216,24 @@ export const processLayoutNodeMap = (executionSummary?: PipelineExecutionSummary
...
 
+function parseStages(stages: ProcessLayoutNodeMapResponse[] | GraphLayoutNode[]): string {
+  const n = stages.length
+  for (let i = n - 1; i >= 0; i--) {
+    const stage = (stages[i] as ProcessLayoutNodeMapResponse)?.stage ?? (stages[i] as GraphLayoutNode)
+
+    if ((stage as ProcessLayoutNodeMapResponse).parallel) {
+      return parseStages((stage as ProcessLayoutNodeMapResponse)?.parallel as GraphLayoutNode[])
+    } else {
+      if (stage.status === 'Skipped') {
```
Comment (vkbansal) [inline src/modules/70-pipeline/utils/executionUtils.ts:227]: please use `isExecutionSkipped` helper from [here](https://github.com/harness/harness-core-ui/blob/85e84789084c51a779dda606ae622c058db227d0/src/modules/70-pipeline/utils/statusHelpers.ts#L135)

**Review by vkbansal — CHANGES_REQUESTED** [review]

File: src/modules/70-pipeline/utils/executionUtils.ts (hunk 1 above)
Comment (AshwinBhatkal-Harness) [reply src/modules/70-pipeline/utils/executionUtils.ts]: Made the change

**Comment by AshwinBhatkal-Harness** [comment]
retrigger typecheck

**Comment by AshwinBhatkal-Harness** [comment]
retrigger typecheck

**Pull request closed without merging**